### PR TITLE
Wrap!

### PR DESCRIPTION
**Pull request by curran**
Implements **tile.wrap(true)** #10. The added code is a verbatim copy of the implementation in [Wrap Test III](http://bl.ocks.org/curran/d497d9c7611ed7278bbcae91454a906c), which serves as the only test for now.

 * [x] Add implementation
 * [x] Add test
 * [x] Update documentation
 * [x] Discuss the 4-element-array representation. Is it fine?

### CODE REVIEW

**Comment by mbostock** [comment]
I don’t like the four-element array. 😦  The problem is the fourth value isn’t really an orthogonal dimension like the other three. And it starts to feel like a long unnamed argument list.

I also don’t like in the current API that you need to multiply *d*[0] and *d*[1] by the tile size (typically 256) to position the tiles. But there are lots of different ways you can position the tiles actually, such as by making them all 1×1 and then applying a transform that scales them by 256 or whatever the current tile scale is… So not sure what the best solution is.

Feels to me like [*x*, *y*, *z*] should use the wrapped *x* always, but then maybe we also expose *d*.x and *d*.y named properties on the array that represent the tile’s position (separate from its identity)? And then there’s the question of whether those values should be premultipled by the tile size. I think the answer there is yes.

**Comment by curran** [comment]
I also don't like the 4 element array. I'll try this:

 * [ ] Make the 3-element array use the wrapped X, as the identity of the tile.
 * [ ] Expose *d.x* and *d.y* as the offsets, premultiplied by the tile size.

That feels a lot better. Thanks for the suggestion!

Before:

```js
  image.enter().append("image")
      .attr("xlink:href", function(d) { return "http://" + "abc"[d[1] % 3] + ".tile.openstreetmap.org/" + d[2] + "/" + d[3] + "/" + d[1] + ".png"; })
      .attr("x", function(d) { return d[0] * 256; })
      .attr("y", function(d) { return d[1] * 256; })
      .attr("width", 256)
      .attr("height", 256);
```

After:

```js
  image.enter().append("image")
      .attr("xlink:href", function(d) { return "http://" + "abc"[d[1] % 3] + ".tile.openstreetmap.org/" + d[2] + "/" + d[0] + "/" + d[1] + ".png"; })
      .attr("width", 256)
      .attr("height", 256)
    .merge(image)
      .attr("x", function(d) { return d.x; })
      .attr("y", function(d) { return d.y; });
```

I ran into some issues with using *enter()* only, so moved the "x" and "y" into a merged enter + update. Here's an updated example that shows this working - [Wrap Test IV](http://bl.ocks.org/curran/881b9f5e0e0c471f81fbfac452ca1438)

**Comment by mbostock** [comment]
You should be able to use *test*.deepEqual to test against nested objects and arrays. Though you might need to do something clever to handle the arrays with the extra *x* and *y* property.

**Comment by curran** [comment]
Done.

**Comment by curran** [comment]
It occurred to me that if the `x` and `y` offsets are multiplied by 256 here, maybe the `tiles.scale` and `tile.translate` values should be modified in the same PR, to eliminate the need for some of the math in this snippet:

```js
function stringify(scale, translate) {
  var k = scale / 256, r = scale % 1 ? Number : Math.round;
  return "translate(" + r(translate[0] * scale) + "," + r(translate[1] * scale) + ") scale(" + k + ")";
}
```

Namely:

 * Divide the scale by 256
 * Multiply the translate by the scale

**Comment by curran** [comment]
I think this PR is good to go. The `stringify` modifications can go in a different PR.

@mbostock I'll go ahead and merge this in myself. Hope you don't mind :smile:

**Comment by curran** [comment]
Here are updated versions of the examples using this new change:

 * [Raster & Vector I (wrap update)](http://bl.ocks.org/curran/b5092379e6eae08b0d36917d21ed4c8d)
 * [Raster & Vector II (wrap update)](http://bl.ocks.org/curran/92140ba6c374af557c18aad824bab387)
 * [Raster & Vector III (wrap update)](http://bl.ocks.org/curran/80962a770d1a8d200eca9a878d52a859)
 * [Raster & Vector IV (wrap update)](http://bl.ocks.org/curran/1689fd22d4cfc55be758daa9d29fd2c4)
 * [Tile by Bounding Box (wrap update)](http://bl.ocks.org/curran/395b9670077d49480ad4d808c7f93eff)

File: test/tile-test.js [hunk 1]
```diff
@@ -18,8 +18,64 @@ tape("tile", function(test) {
...
   test.equal(tiles.translate[1], -5.08203125);
   test.equal(tiles.length, 15);
-  test.deepEqual(tiles[0], [ 1, 5, 4 ]);
-  test.deepEqual(tiles[1], [ 2, 5, 4 ]);
-  test.deepEqual(tiles[2], [ 3, 5, 4 ]);
+
+  var d = [1, 5, 4];
+  d.x = 256;
+  d.y = 5 * 256;
+  test.deepEqual(tiles[0], d);
```
Comment (mbostock) [inline test/tile-test.js]: I recommend creating a helper function like this:

```js
function tile(x, y, z, tx, ty) {
  var d = [x, y, z];
  d.x = tx;
  d.y = ty;
  return d;
}
```

Then you can do:

```js
test.deepEqual(tiles[0], tile(1, 5, 4, 256, 5 * 256));
```

And even better:

```js
test.deepEqual(tiles, [
  tile(1, 5, 4, 1 * 256, 5 * 256),
  tile(2, 5, 4, 2 * 256, 5 * 256),
  tile(3, 5, 4, 3 * 256, 5 * 256),
  …
]);
```

Comment (curran) [inline test/tile-test.js]: Thanks for the suggestion!

Changes for this in PR #16.

**Pull request merged**